File: older/gae-foodtruck-api/src/main/java/com/example/foodtruck/services/IngredientService.java
package com.example.foodtruck.services;

import com.example.foodtruck.models.Ingredient;

public interface IngredientService {

    Ingredient findById(Long id);

    Iterable<Ingredient> getAll();

    Iterable<Ingredient> save(Ingredient ingredient) ;

    Iterable<Ingredient> update(Long id, Ingredient ingredient);

    void delete (Long id);
}

File: api/src/main/kotlin/com/foodtruck/api/services/IngredientService.kt
package com.foodtruck.api.services

import com.foodtruck.api.models.Ingredient

interface IngredientService {
    fun findById(id: Long): Ingredient
    fun getAll(): Iterable<Ingredient>
    fun save(ingredient: Ingredient): Iterable<Ingredient>
    fun update(id: Long, ingredient: Ingredient): Iterable<Ingredient>
    fun delete(id: Long)
}